fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/physical/PhysicalTopNOperator.java

```diff
@@ -110,7 +113,7 @@ public RowOutputInfo deriveRowOutputInfo(List<OptExpression> inputs) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(super.hashCode(), orderSpec, offset, sortPhase, topNType, isSplit);
+        return Objects.hash(super.hashCode(), orderSpec, offset, sortPhase, topNType, isSplit, isEnforced);
```

Comment:
why we need change here?

Reply:
`DecodeContext` record the output dict by map `operatorDecodeInfo<Operator, DecodeInfo>`, and TopN equals&hash is error that find other TopN node when rewrite low-cardinality